src/org/thoughtcrime/securesms/util/Util.java

```diff
@@ -272,15 +272,24 @@ public static float clamp(float value, float min, float max) {
   }
 
   public static void writeTextToClipboard(@NonNull Context context, @NonNull String text) {
-    try {
-      ClipboardManager clipboardManager = (ClipboardManager)context.getSystemService(Context.CLIPBOARD_SERVICE);
-      clipboardManager.setPrimaryClip(ClipData.newPlainText(context.getString(R.string.app_name), text));
```

Comment:
Didn't this code work on older devices?

Reply:
> Didn't this code work on older devices?

seems it didn't, tbh I am not sure, it just was done in two different ways, one cared about android versions and didn't checked for errors and another used one way and checked for errors, so I just used one of them